migrator/migrator.go

```diff
@@ -99,7 +99,10 @@ func (m Migrator) AutoMigrate(values ...interface{}) error {
 			}
 		} else {
 			if err := m.RunWithValue(value, func(stmt *gorm.Statement) (errr error) {
-				columnTypes, _ := m.DB.Migrator().ColumnTypes(value)
+				columnTypes, err := m.DB.Migrator().ColumnTypes(value)
```

Comment:
this `err` is shadowing the previous one, you can remove the `:`

Reply:
It is the local error inside function literal scope, which returns error as well. Looks good to me.